client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/NearCacheService.java

```diff
@@ -170,6 +172,12 @@ public void handleRemovedEvent(ClientCacheEntryRemovedEvent<K> event) {
          invalidate(event.getKey());
       }
 
+      @ClientCacheEntryExpired
```

Comment:
Seems like you're fixing [ISPN-7087](https://issues.jboss.org/browse/ISPN-7087) indirectly here?

Reply:
Yeah let me make a separate commit for it.

Reply:
I didn't realize `EagerNearCacheListener` is still there, but it's private and never used, so I'd rather delete it.
There is a reference to it in the code, but that condition in `NearCacheService.createListener()` is never false.